### PR TITLE
PaperUI: add an OSM source provider for the location widget

### PR DESCRIPTION
**Pull request by htreu**
This provides an OpenStreetMap source provider for the OpenLayers map widget in PaperUI. The widget is used for location items and location configuration parameters.

This depends on ESH with https://github.com/eclipse/smarthome/pull/3730 merged.

Also: update the location pin to use the new openHAB logo.

### CODE REVIEW

File: bundles/org.openhab.ui.paperui/patch/web/js/extensions.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/* This file can be be overridden by fragments to include custom JS code */
+angular.module('PaperUI.extensions', []) //
+
+.service('mapSourceService', function() {
+    return {
+        getMapSource : function() {
+            return new ol.source.OSM()
+        }
+    }
+})
```
Comment (martinvw) [inline bundles/org.openhab.ui.paperui/patch/web/js/extensions.js]: Missing newline

**Comment by NorbertHD** [comment]
Now that  eclipse/smarthome#3730 has been merged, is this PR necessary that the widgets for location configuration parameters work?

I ask because I can not see widgets for PaperUI->Configuration->System->Regional Settings->Location and the location settings in "Astro Sun" and "Astro Moon".

**Comment by kaikreuzer** [comment]
Correct, for the moment there is only a text input field.

Before this PR can be merged, we need to get approval from OSM to use their tile server in openHAB.

**Comment by htreu** [comment]
The default tile server URL for the OSM source is `https://{a-c}.tile.openstreetmap.org/{z}/{x}/{y}.png`, where `x,y,z` are the coordinates and I guess `a-c` are different servers (for which I don't know how they are chosen).

**Comment by htreu** [comment]
For the new custom ImageTile load function the framework has to provide a new OpenLayers API. So this PR now depends on ESH https://github.com/eclipse/smarthome/pull/5059.

**Pull request merged**